[Fix] 공지사항 Api 수정

Young-Yeong opened the pull request.

Young-Yeong committed:
Fix : Point 관련 누락 파일 추가

- PayRepository
    - Jpa 적용

Related to : #92

Young-Yeong committed:
Fix : 공지사항 Api 수정

- NoticeContentResponseDto
    - 이름 변경으로 인한 삭제

- NoticeController
    - getNoticesByBranchName
        - Integer count 추가
    - getNoticeContentByNoticeId
        - Response값 수정

- NoticeListResponseDto
    - 공지사항 리스트 Response Dto

- NoticeRepositoryCustom
    - getNoticesByBranchName
        - 반환값 수정

- NoticeRepositoryImpl
    - getNoticesByBranchName
        - query문 limit추가

- NoticeResponseDto
    - 공지사항 상세조회 Response Dto

- NoticeService
    - getNoticesByBranchName
        - Integer count 추가
    - getNoticeContentByNoticeId
        - 반환값 수정

Related to : #93

Capoomaru commented on src/main/java/com/yu/yurentcar/domain/branch/dto/NoticeListResponseDto.java:
```diff
@@ -0,0 +1,24 @@
+    private String photoUrl;
+    private String title;
+    private String description;
+    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss", timezone = "Asia/Seoul")
+    private LocalDateTime startDate;
+    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss", timezone = "Asia/Seoul")
+    private LocalDateTime finishDate;
+    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss", timezone = "Asia/Seoul")
+    private LocalDateTime modifiedAt;
```
@JsonFormat 활용 좋아요 👍

Capoomaru commented on src/main/java/com/yu/yurentcar/domain/branch/dto/NoticeListResponseDto.java:
```diff
@@ -0,0 +1,24 @@
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class NoticeListResponseDto {
```
리스트를 위한 DTO 인거면, 명칭이 `NoticeSummaryDto` 와 같은 명칭을 쓰고
`NoticeListResponseDto`는 `List<NoticeSummaryDto>` 를 가지는 구조가 더 좋아보여요.

Capoomaru commented on src/main/java/com/yu/yurentcar/domain/branch/repository/NoticeRepositoryImpl.java:
```diff
@@ -22,21 +22,21 @@ public class NoticeRepositoryImpl implements NoticeRepositoryCustom {
+                                .where(branch.branchName.eq(branchName).and(branch.siDo.eq(siDo))))
                         .and(notice.finishDate.gt(LocalDateTime.now()).or(notice.finishDate.isNull())))
                 .orderBy(notice.modifiedAt.desc())
+                .limit(count == 0 ? Long.MAX_VALUE : count)
```
MAX로 매핑하는 것은 쿼리 성능 상 위험해요. limit 문항을 따로 채워주세요.
```java
//개선 코드
JPAQuery<NoticeListResponseDto> query = queryFactory
                .select(Projections.constructor(NoticeListResponseDto.class, notice.noticeId, notice.photoUrl, notice.title, notice.description, notice.startDate, notice.finishDate, notice.modifiedAt))
                .from(notice)
                .where(notice.branch.eq(queryFactory
                                .selectFrom(branch)
                               .where(branch.branchName.eq(branchName).and(branch.siDo.eq(siDo))))
                        .and(notice.finishDate.gt(LocalDateTime.now()).or(notice.finishDate.isNull())))
                .orderBy(notice.modifiedAt.desc());
if(count != 0)
    query = query.limit(count);

return query.fetch();
```

(그리고 실제 limit는 최대 Integer.MAX_VALUE에요)
<img width="575" alt="image" src="https://github.com/YU-RentCar/YURentCar-be/assets/44056518/925b916c-d2a0-4e32-9282-c6adee9e493e">

Capoomaru commented on src/main/java/com/yu/yurentcar/domain/branch/service/NoticeService.java:
```diff
@@ -36,17 +36,16 @@ public List<NoticeResponseDto> getNoticesByBranchName(String province, String br
+    public NoticeResponseDto getNoticeContentByNoticeId(Long noticeId) {
         Optional<Notice> notice = noticeRepository.findById(noticeId);
-        if (notice.isEmpty()) throw new RuntimeException();
-        return NoticeContentResponseDto.builder()
+        if (notice.isEmpty()) throw new RuntimeException("없는 공지사항입니다.");
```
새로 작성되는 코드는 아래와 같이 처리해서 `.get()`의 반복을 줄여주세요
```java
Notice notice = noticeRepository.findById(noticeId).orElseThrow(() => new RuntimeException("없는 공지사항입니다."));
```

Capoomaru reviewed (approved):
고생하셨습니다!

Young-Yeong committed:
Fix : #94 수정

- NoticeController
    - getNoticesByBranchName
        - 반환값 변경

- NoticeRepositoryCustom
    - getNoticesByBranchName
        - 반환값 변경

- NoticeRepositoryImpl
    - getNoticesByBranchName
        - query문 수정 및 반환값 변경

- NoticeService
    - getNoticesByBranchName
        - get() 반복 개선

- NoticeSummaryDto
    - 이름 변경

Related to : #94

Capoomaru reviewed (approved):
LGTM 👍
